bin/user/purpleair.py

```diff
@@ -232,6 +244,7 @@ def __init__(self, engine, config_dict):
         self.config_dict = config_dict.get('PurpleAirMonitor', {})
         try:
             self.config_dict['hostname']
+            self.config_dict['api_key']
```

Comment:
@xslima00 this requires the api_key to be provided even when fetching from the local device when it is unnecessary. I'd recommend we make api_key only required when the hostname.isnumeric() like the other test for if we are fetching from the purpleair website.

Reply:
@bakerkj of course you are right, last commit makes this optional